test/IntegrationTest.ts

```diff
@@ -0,0 +1,366 @@
+        erc20.balanceOf(ainpayment.address),
+        ethers.provider.getBalance(ainpayment.address)
+      ]);
+
+      expect(ownerInitAIN.add(ainpaymentInitAIN)).to.equal(ownerUpdatedAIN);
+      expect(ownerUpdatedETH.sub(ainpaymentInitETH)).to.lessThan(ownerInitETH);
+      console.log("Before balance of AinPayment(AIN:ETH): ", ainpaymentInitAIN.toString(), ainpaymentInitETH.toString());
+      console.log("After balance of AinPayment(AIN:ETH): ", ainpaymentUpdatedAIN.toString(), ainpaymentUpdatedETH.toString());
+    });
+    it("should not withdraw AINPayment except owner", async function () {
+      // minter tries to withdraw all AINPayment
+      const [minterInitAIN, minterInitETH] = await Promise.all([
+        erc20.balanceOf(await minter.getAddress()),
+        ethers.provider.getBalance(await minter.getAddress())
+      ]);
+      const [ainpaymentInitAIN, ainpaymentInitETH] = await Promise.all([
+        erc20.balanceOf(ainpayment.address),
+        ethers.provider.getBalance(ainpayment.address)
+      ]);
+      
+      await expect(ainpayment.connect(minter).withdraw(1)).to.be.reverted;
+      await expect(ainpayment.connect(minter).withdrawAll()).to.be.reverted;
+      
+    });
+
+    it("should destroy AINPayment", async function () {
+      // destroy AINPayment
+      const [ownerInitAIN, ownerInitETH] = await Promise.all([
+        erc20.balanceOf(await owner.getAddress()),
+        ethers.provider.getBalance(await owner.getAddress())
+      ]);
+      const [ainpaymentInitAIN, ainpaymentInitETH] = await Promise.all([
+        erc20.balanceOf(ainpayment.address),
+        ethers.provider.getBalance(ainpayment.address)
+      ]);
+      
+      //destruct AINPayment
+      await expect(ainpayment.connect(owner).destruct("NOTDELETE")).to.be.reverted; // revert if not "DELETE"
+      await expect(ainpayment.connect(owner).destruct("DELETE")).not.to.be.reverted;
+      
+      const [ownerUpdatedAIN, ownerUpdatedETH] = await Promise.all([
+        erc20.balanceOf(await owner.getAddress()),
+        ethers.provider.getBalance(await owner.getAddress())
+      ]);
+      const [ainpaymentUpdatedAIN, ainpaymentUpdatedETH] = await Promise.all([
+        erc20.balanceOf(ainpayment.address),
+        ethers.provider.getBalance(ainpayment.address)
+      ]);
+
+      expect(ownerInitAIN.add(ainpaymentInitAIN)).to.equal(ownerUpdatedAIN);
+      expect(ownerUpdatedETH.sub(ainpaymentInitETH)).to.lessThan(ownerInitETH);
+      console.log("Before balance of AinPayment(AIN:ETH): ", ainpaymentInitAIN.toString(), ainpaymentInitETH.toString());
+      console.log("After balance of AinPayment(AIN:ETH): ", ainpaymentUpdatedAIN.toString(), ainpaymentUpdatedETH.toString());
+    
+    });
+
+    it("should not destroy AINPayment except owner", async function () {
+      // destroy AINPayment from minter
+      const [minterInitAIN, minterInitETH] = await Promise.all([
```

Comment:
both minter/ainpayment balance variables are unused. need comparison or removed?

Reply:
Same as above. I'll remove it thank you!